source/common/upstream/load_balancer_impl.cc

```diff
@@ -228,9 +228,13 @@ void ZoneAwareLoadBalancerBase::resizePerPriorityState() {
 bool ZoneAwareLoadBalancerBase::earlyExitNonLocalityRouting() {
   // We only do locality routing for P=0.
   HostSet& host_set = *priority_set_.hostSetsPerPriority()[0];
+  if (host_set.healthyHostsPerLocality().get().size() < 2) {
```

Comment:
What's the reasoning behind this change? Can you add more comments?

Reply:
I think this better matches the documented semantics for `lb_local_cluster_not_ok`:

"Local host set *is not set* or it is panic mode for local cluster"

Reply:
OK makes sense, thanks.